### DISCUSSION TITLE
WinThumbsPreloader 2.0.0 Beta 1

### DISCUSSION THREAD

**Discussion by Mfarooq360** [Mfarooq360/WinThumbsPreloader-V2#3] [Announcements]
***This is a Beta version, which may not be thoroughly tested and may have some bugs, feel free to report them along with any other suggestions so I can help to improve the release.***

*This changelog will be very incomplete, full changelog will be in the release version.*
## Major Changes:
Updated to .NET 6
Added a button to select a folder to preload or to preload recursively while holding shift
Added a settings form, schedule form, and cache form (More info in notes section)
Allow multiple paths to be processed through the command line
Made command line arguments far more flexible and forgiving
Added toggle for preloading icons for folders with specified thumbnail extensions
Thread count can now be specified through the command prompt or set in the settings
Thumbnail cache can now be backed up after the preloader is done when auto backup thumbnails are enabled.
Administrator mode can now be toggled
Added beta logic to updater

## Planned Features and Changes:
Add a way to modify amount of folders to preload at once using a similar method to selecting the thread count
Add a right click context menu for the empty space of a folder
Add a way to automatically generate new audio file thumbnails using folder.jpg or cover.jpg files
Potentially make backup and restore intervals modifiable
Fix various bugs and clean up code and codebase

## Notes:
There are various new forms added to the program, each with their own various abilities.

The **Settings Form** allows you to change thumbnail extensions without having to open notepad and lets you select whether to preload icons for folders that contain files with the specified thumbnail extensions. There are also toggles for running as administrator and for multithreading along with being able to select the thread count.

The **Schedule Form** allows you to run WinThumbsPreloader on selected drives for every specified interval. You can select to have it run on idle along with being able to change how many threads it can use. You can also disable and enable it with a button in order to keep your settings.

The **Cache Form**, which is the strongest part of this new update, allows you to backup and restore the thumbnail cache, along with being able to be alerted when the cache resets on startup of the application. The sizes of both the backup cache and the thumbnail cache are displayed in the cache form. 
There are also toggles to automatically backup and restore the thumbnail cache, which checks every 5 seconds. The auto backup will occur when the thumbnail cache is higher than the backup and is stable at the current size. The auto restore will occur anytime the thumbnail cache gets deleted. You can also manually delete both the backup or the thumbnail cache by holding shift before pressing the open backup or cache buttons. 
When minimizing the cache form, the program will go into the tray, where it can auto backup and restore out of the way of the taskbar. The tray icon for WinThumbsPreloader can be right clicked in order to make changing settings easier. You can toggle if the cache form starts with Windows, which allows it to startup minimized in the tray. The program can also startup minimized in the tray manually if you pass the command line argument "-startminimized"

The command line has been made to take paths in a far more flexible way. You are now able to choose whether or not you would like to use quotes, commas, or even spaces between paths! This helps to make adding multiple paths to the command arguments far easier.

<hr /><em>This discussion was created from the release <a href='https://github.com/Mfarooq360/WinThumbsPreloader-V2/releases/tag/v2.0.0-beta1'>WinThumbsPreloader 2.0.0 Beta 1</a>.</em>

**Comment by moxin3** [comment]
when Beta 5 or new version will be released its been months now? waiting for new version

**Reply by Mfarooq360** [reply]
Hi, I'm almost done with the changes I'm making. For the next beta version, we're gonna try to find the causes of any bugs by adding logging, which should hopefully help people who are having trouble with getting the program to run correctly. I'm estimating the next beta to hopefully be put out before the end of this month.

**Reply by moxin3** [reply]
Thanks looking forward and excited for new version, Happy Ramadan brother

**Reply by moxin3** [reply]
March is about to end, i checkup the page daily for new version cox old version get my thumbnails deleted after some time and i have to re-do all thumbs again

**Reply by VL4DST3R** [reply]
Trust me, I feel for you moxin but life gets in the way, I'm sure mutahar will work more on it eventuallly.

ATM I'd recommend you trying older versions to see if they perform better for you, even if they may be slower/have other bugs. In my case I feel lately I had more success with `WinThumbsPreloader1.1.0` and is what I've been using for the last few days when I needed thumbs generated.

**Comment by moxin3** [comment]
> Trust me, I feel for you moxin but life gets in the way, I'm sure mutahar will work more on it eventuallly.
> 
> ATM I'd recommend you trying older versions to see if they perform better for you, even if they may be slower/have other bugs. In my case I feel lately I had more success with `WinThumbsPreloader1.1.0` and is what I've been using for the last few days when I needed thumbs generated.

thanks its good to know someone also excited for new release, yes i ll give it a try the older version

**Reply by Mfarooq360** [reply]
Hi, I have to delay the release of the next beta for a bit as the changes being added is taking a long while due to the complexity of the features being added along with many bugs being found. 

The main thing taking a long while is adding logging which should help us in finding out any problems or other bugs in the code and perhaps may help us find what's causing thumbnail cache resets potentially.

The new update won't fix this problem since it's a complex issue which may be a problem with windows itself, but logging may help us determine if it's instead a problem with the way the program preloads thumbnails.

**Comment by Korb** [comment]
>  Added a button to select a folder to preload or to preload recursively while holding **shift**

It would be very useful to include this in the basic information about the app's capabilities in the `README.md` file. And on the app's main screen, of course. Because right now, it only displays the following text:

> WinThumbsPreloader is up to date
> To use this app, press "Preload" or right click on a folder to call the context menu and select WinThumbsPreloader > Preload thumbnails in the menu.
> This app can also be launched from the command line:
> Usage:
> `WinThumbsPreloader.exe [-r -s] <path>`
> Options:
> `-r` Recursively preload thumbnails in nested folders.
> `-s` Silent mode - don't show a progress dialog.
> `-m` / `-m<n>` / `-m:<n>` Multi-Threaded mode with optional thread count
> File extensions for thumbnail preloading can be modified in the Settings.
> Project page: `https://github.com/Mfarooq360/WinThumbsPreloader-V2`
> Copyright (c) 2018 Dmitry Bruhov
> Copyright (c) 2025 Mutahar Farooq
> `Preload` / `Settings` / `License` / `Close`
